Modules/_struct.c

```diff
@@ -1390,8 +1360,6 @@ prepare_s(PyStructObject *self)
             num = c - '0';
             while ('0' <= (c = *s++) && c <= '9')
                 num = num*10 + (c - '0');
-            if (c == '\0')
```

Comment:
IMHO, this part is another round same as the previous round but without checks. So `c == '\0'` can't be true anyway otherwise the previous round should have failed. Even if it could be true, does simply break the loop a right behaviour?